zeebe/gateway-protocol/vacuum-rules/verifyKeyTypesSchemas.js

```diff
@@ -0,0 +1,13 @@
+function runRule(input) {
+  // `input` is an entire schema object with all properties.
+  //   Identify all properties that end with Key, and are not strings.
+  return Object.entries(input)
+    .filter(([name, schema]) => {
+      return /Key$/.test(name) && schema.type !== "string";
+    })
+    .map((_) => {
+      return {
+        message: "`...Key` properties must be of type `string`.",
+      };
```

Comment:
This specific rule is also a little funny, in that it associates the line numbers with a schema, rather than the individual property that violates the rule. This is again because of the structure of the document, specifically that the name of the property is the key of the object. I was unable to find a JSONPath query that gave me both the property name (object key) and the data type (a property of that object).

To a user, this shows as the line numbers in the output pointing them at the schema rather than the specific property.

Reply:
Could we display the actual property names in the message?
I think it would be nicer.

Reply:
Good catch, @danielkelemen 👍